geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/ConvertToGeoFile.scala

```diff
@@ -0,0 +1,238 @@
+              case ExportFormat.Gml2    => GmlExporter.gml2(stream)
+              case ExportFormat.Gml3    => GmlExporter(stream)
+              case ExportFormat.Json    => new GeoJsonExporter(stream)
+              case ExportFormat.Leaflet => new LeafletMapExporter(stream)
+              case ExportFormat.Tsv     => DelimitedExporter.tsv(stream, headers)
+              // shouldn't happen unless someone adds a new format and doesn't implement it here
+              case _ => throw new NotImplementedError(s"Export for '$format' is not implemented")
+            }
+            export(exporter)
+          }
+        })
+      } else {
+        val dir = Files.createTempDirectory("gm-nifi")
+        try {
+          val file = new File(dir.toFile, s"export.${format.extensions.head}")
+          val exporter = format match {
+            case ExportFormat.Orc     => new OrcFileSystemExporter(file.getAbsolutePath)
+            case ExportFormat.Parquet => new ParquetFileSystemExporter(file.getAbsolutePath)
+            // case ExportFormat.Shp     => new ShapefileExporter(file) // TODO zip up dir??
+            // shouldn't happen unless someone adds a new format and doesn't implement it here
+            case _ => throw new NotImplementedError(s"Export for '$format' is not implemented")
+          }
+          export(exporter)
+          output = session.importFrom(file.toPath, false, output)
+        } finally {
+          PathUtils.deleteRecursively(dir)
+        }
+      }
+
+      if (result != null) {
+        output = session.putAttribute(output, "geomesa.convert.successes", result.success.toString)
+        output = session.putAttribute(output, "geomesa.convert.failures", result.failure.toString)
+      }
+
+      val basename =
+        Option(input.getAttribute("filename")).map(FilenameUtils.getBaseName).getOrElse(UUID.randomUUID().toString)
+      output = session.putAttribute(output, "filename", s"$basename.${format.extensions.head}")
+
+      session.transfer(output, Relationships.SuccessRelationship)
+      session.remove(input)
+    } catch {
+      case NonFatal(e) =>
+        logger.error(s"Error converting file: ${e.getMessage}", e)
+        session.transfer(input, Relationships.FailureRelationship)
+        if (output != null) {
+          session.remove(output)
+        }
+    }
+  }
+}
+
+object ConvertToGeoFile {
+
+  import scala.collection.JavaConverters._
+
+  val Formats = Seq(
+    ExportFormat.Arrow,
+    ExportFormat.Avro,
+    ExportFormat.Bin,
+    ExportFormat.Csv,
+    ExportFormat.Gml2,
+    ExportFormat.Gml3,
+    ExportFormat.Json,
+    ExportFormat.Leaflet,
+    ExportFormat.Orc,
+    ExportFormat.Parquet,
+    // TODO ExportFormat.Shp,
+    ExportFormat.Tsv
+  )
+
+  /**
+   * Gets the hints to inform an arrow export.
+   *
+   * Currently we:
+   *   <ul>
+   *     <li>exclude fids</li>
+   *     <li>dictionary encode all string fields</li>
+   *     <li>TODO sort by date</li>
```

Comment:
Yeah, we definitely need a ticket for this option.  I'd opine that we should do it before the 3.2.0 release.  (It ought to be easy, right?)

Reply:
mmm, i'm not sure. we don't support it in the CLI tools exporter, so we'd have to implement it there. i assume it wasn't implemented there b/c it's not trivial, or I probably would have done it at the time. again, wasn't sure if anyone would ever use this so didn't want to spend more time on it. i think it would likely be possible to pre-sort your input if you really needed to get out a sorted arrow file.

here's the block in question: https://github.com/locationtech/geomesa/blob/main/geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/export/formats/ArrowExporter.scala#L57